<launch>
    <arg name="map_frame"    default="map" doc="target frame for occupancy grid map"/>
    <arg name="cloud_frame"  default="lidar" doc="source frame for occupancy grid map"/>
    <arg name="map_width"    default="1000" doc="width of map"/>
    <arg name="map_height"   default="1000" doc="height of the map"/>
    <arg name="map_res"      default="0.01" doc="resolution of map"/>
    <arg name="verbose"      default="false" doc="whether to print stuff or not"/>

    <!-- occupancy grid launcher -->
	<node name="grid_generator" pkg="dragoon_simulation" type="grid_generator" output="screen">
        <param name="map_frame"    value="$(arg map_frame)"/>
        <param name="cloud_frame"  value="$(arg cloud_frame)"/>
        <param name="map_width"    value="$(arg map_width)"/>
        <param name="map_height"   value="$(arg map_height)"/>
        <param name="map_res"      value="$(arg map_res)"/>
        <param name="verbose"      value="$(arg verbose)"/>
    </node>

    <!-- static map frame publisher -->
    <node name="static_map_broadcaster" pkg="dragoon_simulation" type="map_frame_broadcaster" output="screen"/>
</launch>